osbs/tekton.py

```diff
@@ -498,14 +498,19 @@ def get_error_message(self):
 
             if 'steps' in stats['status']:
                 for step in stats['status']['steps']:
-                    exit_code = step['terminated']['exitCode']
-                    if exit_code == 0:
-                        continue
-
-                    reason = step['terminated']['reason']
-                    err_message += f"task step '{step['name']}' failed with exit " \
-                                   f"code: {exit_code} " \
-                                   f"and reason: '{reason}'"
+                    if 'terminated' in step:
+                        exit_code = step['terminated']['exitCode']
+                        if exit_code == 0:
+                            continue
+
+                        reason = step['terminated']['reason']
+                        err_message += f"task step '{step['name']}' failed with exit " \
+                                       f"code: {exit_code} " \
+                                       f"and reason: '{reason}'"
+                    else:
+                        err_message += f"task step '{step['name']}' is missing 'terminated' key " \
```

Comment:
If we logged something here about the taskRun status, could it help us figure  out why this is happening? Or is this one of those things that are logged only in koji-containerbuild that we'll never see?

Reply:
task run status doesn't contain anything useful apart from succeeded or failed, if you want details you need to check steps, which is this code doing